src/Infrastructure.Styles.Analyzer/RMSYN1102SeparateExitConditionsCodeFixProvider.cs

```diff
@@ -0,0 +1,213 @@
+    }
+
+    private static SyntaxNode ReplaceIllegalStatement (SyntaxNode syntaxRoot, TextSpan diagnosticSpan)
+    {
+      (SyntaxNode Node, IfStatementSyntax IfStatement) illegalExpression;
+      try
+      {
+        illegalExpression =CheckIfIllegalExpression(syntaxRoot.FindNode(diagnosticSpan));
+      }
+      catch
+      {
+        return syntaxRoot;
+      }
+
+      var syntaxWrapper = CreateSyntaxWrapper(illegalExpression.IfStatement);
+      
+      //need to remove else otherwise there are else clauses after every if and that's not desirable
+      var syntaxWrapperWithoutElse = syntaxWrapper.Copy();
+      syntaxWrapperWithoutElse.RemoveElseClause();
+
+      var syntaxNodes = new List<IfAndElseIfStatementSyntaxWrapper>();
+
+      FillSyntaxNodeListWithStatements(syntaxNodes, syntaxWrapperWithoutElse,
+        illegalExpression.Node);
+
+      //time to handle the previously ignored else clauses
+      if (syntaxWrapper.HasElseClause)
+        HandleElseClauses(syntaxWrapper, syntaxNodes);
+
+      return BuildRoot(syntaxRoot, syntaxNodes, syntaxWrapper);
+    }
+
+    private static void HandleElseClauses (
+      IfAndElseIfStatementSyntaxWrapper syntaxWrapper,
+      List<IfAndElseIfStatementSyntaxWrapper> syntaxNodes)
+    {
+      if (syntaxWrapper.ElseClauseHasIfStatement())
+      {
+        var heldIfStatement = syntaxWrapper.ElseClause!.Statement as IfStatementSyntax;
+        var elseClauseSyntaxWrapper = CreateSyntaxWrapper(heldIfStatement!);
+        FillSyntaxNodeListWithStatements(syntaxNodes, elseClauseSyntaxWrapper, heldIfStatement!.Condition);
+      }
+      else
+      {
+        syntaxNodes.Add(new IfAndElseIfStatementSyntaxWrapper(syntaxWrapper.ElseClause!, null));
+      }
+
+      ReintroduceParentElseClauses(syntaxNodes);
+    }
+
+    private static IfAndElseIfStatementSyntaxWrapper CreateSyntaxWrapper (IfStatementSyntax ifStatement)
+    {
+      IfAndElseIfStatementSyntaxWrapper blueprint;
+      if (ifStatement.Parent.IsKind(SyntaxKind.ElseClause))
+        blueprint = new IfAndElseIfStatementSyntaxWrapper(ifStatement.Parent, ifStatement.Else);
+      else
+        blueprint = new IfAndElseIfStatementSyntaxWrapper(ifStatement, ifStatement.Else);
+
+      return blueprint;
+    }
+
+    private static SyntaxNode BuildRoot (
+      SyntaxNode oldRoot,
+      List<IfAndElseIfStatementSyntaxWrapper> syntaxNodes, 
+      IfAndElseIfStatementSyntaxWrapper oldNode)
+    {
+      BuildProperStatementList(syntaxNodes);
+
+      //ReplaceNode() can only replace an ifNode by a list if the list does not include else statements.
+      //Else if statements can only be in the list if it's count is 1
+      if (syntaxNodes.Count == 1)
+        return oldRoot.ReplaceNode(oldNode.IfOrElseIfStatement, syntaxNodes.First().IfOrElseIfStatement);
+      return oldRoot.ReplaceNode(oldNode.IfOrElseIfStatement,
+        syntaxNodes.Select(s => s.IfOrElseIfStatement));
```

Comment:
The line hard wrap option is probably another one that should be disabled in the settings. We do not want to have lines that are too long but automatic line wrap is kind of annoying. We also have a higher than default setting for the line character count before a break is forced

```suggestion
        return oldRoot.ReplaceNode(oldNode.IfOrElseIfStatement, syntaxNodes.First().IfOrElseIfStatement);

      return oldRoot.ReplaceNode(oldNode.IfOrElseIfStatement, syntaxNodes.Select(s => s.IfOrElseIfStatement));
```